### DISCUSSION TITLE
【要望】新規質問の投稿時にプレビューに誘導して欲しい

### DISCUSSION THREAD

**Discussion by nak435** [increments/qiita-discussions#607] [要望・アイデア・不具合]
いつもお世話になっています。

新規の質問の投稿時において、提案・要望があります。

Qiitaに質問されてくる方は、このためにアカウントを登録され、初めてQiitaに投稿される方が非常に多いと思われます。
そんなQiita初心者の方は、大量のソースコードを本文に貼ってくる方が非常に多いです。

新規質問時のテンプレートには、親切にコードブロックの書き方の説明があるのですが、Markdownに慣れていないが為に、使いこなせないということだと思います。

Markdownに慣れておらず、致し方ないところもあると思いますが、投稿前にプレビューで確認すれば、ご自身でもインデントがなくなっているとか、気付けるチャンスがあると思います。しかし、プレビューで確認すること自体を分かっていないと思われるため、

**投稿前に 一度は必ずプレビューに誘導する**仕掛けをご検討お願いします。

投稿時の下記のダイアログに、「投稿前にプレビューで確認しましたか？」というメッセージと、「プレビュー」ボタンを置くだけでも効果があると思われます。
（「投稿前にプレビューで確認しましたか？」というチェックボックスをおいて、チェックしないと「投稿」ボタンが活性化しないという方法もあると思いますが、一度もプレビューしていない場合は、強制的にプレビューに遷移する荒技でも良いかと思うぐらいです）

![スクリーンショット 2024-01-26 10 38 50](https://github.com/increments/qiita-discussions/assets/42625437/df8117e1-f368-443c-adf0-0bea7a532926)

今は、「コードは正しくコードブロックに書きましょう」とコードブロックの書き方のURLを付けて、質問に回答していましたが、この手間が省けます。

ぜひ、ご検討のほど、よろしくお願いします。

**Comment by tanaka-asami415** [comment]
@nak435 
ご提案ありがとうございます🙏

回答してくださる方の負荷をできるだけ少なくしたいと考えているので、何かしらの改善をしたいと思いますが
現在、画面幅が一定数以上ある場合、初期値ではエディタ画面の右半分にプレビューが表示されるようになっているため、既に投稿前にプレビューを確認している方も少なくないと思います。
そのため、投稿前に必ずプレビューに誘導した場合でも、期待する効果があまり見込めない可能性もあると考えています。

ご提案いただいた内容も合わせて、他の方法でも改善ができないか検討したいと思います🙇

**Reply by nak435** [reply]
> そのため、投稿前に必ずプレビューに誘導した場合でも、期待する効果があまり見込めない可能性もあると考えています。

そうなら、確かに期待薄ですね。
プレビューを確認していながら、そのままで投稿する方の心理はなんなのでしょうか。
コードブロックを知らないから、醜いくと思いながら投稿しているのでしょうかね？？？

**Reply by tanaka-asami415** [reply]
@nak435 
右半分にプレビューが表示されない画面幅の狭い方もいるため、そういう方に向けては、プレビュー確認を案内する必要があると考えています。

> コードブロックを知らないから、醜いくと思いながら投稿しているのでしょうかね？？？

プレビューを見ていても、崩れている事に気づいていないという可能性もあるかもしれません…。
現在表示されているテンプレートの例に、NG例など載せたり、nak435さんに案内いただいているような「コードは正しくコードブロックに書きましょう」という記載を行うなど、もっと初心者の方でも、活用しやすいテンプレートにできないかも検討したいと思います🙏
